lib/dal/src/attribute/value/dependent_value_graph.rs

```diff
@@ -163,52 +158,46 @@ impl DependentValueGraph {
-            // the ancestry (aka frames/nested frames)
-            if relevant_apas.is_empty() {
-                if let ValueIsFor::OutputSocket(_) = value_is_for {
-                    let maybe_values_depend_on =
-                        match Component::find_inferred_values_using_this_output_socket(
-                            ctx,
-                            current_attribute_value_id,
-                        )
-                        .await
-                        {
-                            Ok(values) => values,
-                            // When we first run dvu, the component type might not be set yet.
-                            // In this case, we can assume there aren't downstream inputs that need to
-                            // be queued up.
-                            Err(ComponentError::ComponentMissingTypeValueMaterializedView(_)) => {
-                                vec![]
-                            }
-                            Err(err) => return Err(AttributeValueError::Component(Box::new(err))),
-                        };
-
-                    for input_socket_match in maybe_values_depend_on {
-                        let source_component =
-                            Component::get_by_id(ctx, input_socket_match.component_id)
-                                .await
-                                .map_err(|e| AttributeValueError::Component(Box::new(e)))?;
-                        let destination_component_id =
-                            AttributeValue::component_id(ctx, current_attribute_value_id).await?;
-                        let destination_component =
-                            Component::get_by_id(ctx, destination_component_id)
-                                .await
-                                .map_err(|e| AttributeValueError::Component(Box::new(e)))?;
-
-                        // Both "deleted" and not deleted Components can feed data into
-                        // "deleted" Components. **ONLY** not deleted Components can feed
-                        // data into not deleted Components.
-                        if destination_component.to_delete()
-                            || (!destination_component.to_delete() && !source_component.to_delete())
-                        {
-                            work_queue.push_back(input_socket_match.attribute_value_id);
-                            dependent_value_graph.value_depends_on(
-                                input_socket_match.attribute_value_id,
-                                current_attribute_value_id,
-                            );
+            // Also check if this value is an output socket as the attribute value
+            // might have implicit dependendcies based on the ancestry (aka frames/nested frames)
+            // note: we filter out non-deleted targets if the source component is set to be deleted
+            if let ValueIsFor::OutputSocket(_) = value_is_for {
```

Comment:
Do we no longer need to check if apas.is_empty?

Reply:
correct! if the attribute value for an output socket changes, we want to queue inferred AND explicit connections (so drawing edges and having components inside of down frames for example). Doing this check is what was preventing data from flowing to both components in a frame and components connected to the frame's output socket